src/polygon_filter.cpp

```diff
@@ -488,25 +492,31 @@ bool StaticLaserScanPolygonFilter::update(const sensor_msgs::LaserScan& input_sc
     tf::TransformListener transform_listener;
 
     std::string error_msg;
+    ROS_DEBUG(
+      "[StaticLaserScanPolygonFilter] waitForTransform %s -> %s",
+      polygon_frame_.c_str(), input_scan.header.frame_id.c_str()
+    );
     bool success = transform_listener.waitForTransform(
-        input_scan.header.frame_id, polygon_frame_,
-        input_scan.header.stamp + ros::Duration().fromSec(input_scan.ranges.size() * input_scan.time_increment),
-        ros::Duration(1.0), ros::Duration(0.01), &error_msg);
+      input_scan.header.frame_id, polygon_frame_,
+      ros::Time(),       // No restrictions on transform time. It is static.
+      ros::Duration(transform_timeout_),
+      ros::Duration(0),  // This setting has no effect
+      &error_msg
+    );
 
     if (!success) {
       ROS_WARN("[StaticLaserScanPolygonFilter] Could not get transform, ignoring laser scan! %s", error_msg.c_str());
```

Comment:
Could you switch this to ROS_WARN_THROTTLE, maybe with a 1 per second rate? Don't want to spam the console.

Reply:
Actually can use `ROS_WARN_THROTTLE_NAMED("StaticLaserScanPolygonFilter", ...)`

Reply:
Done. Updated all logging that had custom prefix accordingly